drivers/dac/Kconfig.stm32

```diff
@@ -0,0 +1,35 @@
+# DAC configuration options
+
+# Copyright (c) 2020 Libre Solar Technologies GmbH
+#
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig DAC_STM32
+	bool "STM32 DAC driver"
+	depends on SOC_FAMILY_STM32
+	help
+	  Enable the driver implementation for the stm32xx DAC
+
+if DAC_STM32
+
+config DAC_1
```

Comment:
Instead of doing this we can us DT_INST_.. defines in the driver.

Reply:
I'm currently implementing this change, completely removing the `CONFIG_DAC_x` Kconfig settings.
However, I'm not sure how to implement pinmux.c in the board file. It's currently
```
#ifdef CONFIG_DAC_1
	{STM32_PIN_PA4, STM32L0_PINMUX_FUNC_PA4_DAC_OUT1},
#endif /* CONFIG_DAC_1 */
```
Should I use `DT_INST_0_ST_STM32_DAC` again? I think that's confusing because the DAC is named dac1 in the .dts file (aligned with naming in STM32, which starts from 1). I could also use `#ifdef DT_ALIAS_DAC_1_LABEL`, but this doesn't seem very nice either, as it would require to define an alias.

Reply:
@martinjaeger, `DT_INST_0_ST_STM32_DAC` should indeed not be used in case of pinmux file. You should keep using `CONFIG_DAC_1` in this case (soon to be replaced by a new `DT_NODELABEL_FOO` macro.
This is an on going change (https://github.com/zephyrproject-rtos/zephyr/issues/19285 provides an overview). Sorry for the confusion it creates in your  porting work.

Reply:
Alright, so I'll revert the changes again and keep the Kconfig settings.

Reply:
So are there driver reverts that are needed?

Reply:
No, should be fine as it is in this PR.